scripts/coccinelle/unsigned_assignment.cocci

```diff
@@ -0,0 +1,40 @@
+// Find assignments to unsigned variables and add an 'U' to the value
+// Copyright (C) 2018 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+virtual patch
+virtual report
+
+@r_unsigned@
+typedef uint8_t, uint16_t, uint32_t, uint64_t;
+{unsigned char, unsigned short, unsigned int, uint8_t, uint16_t, uint32_t, uint64_t} v;
+constant C;
+position p;
+@@
+
+  v = C@p
```

Comment:
Use:
```
(
v = i 
|
v = C 
)
```
where i is an identifier
and get rid of the isdigit() check as @JuliaLawall suggested!

We use python/Ocaml scripting when we have no options SmPL grammar.

Reply:
This modification works ok, but...